pkg/process/metadata/workloadmeta/workloadmeta.go

```diff
@@ -29,19 +29,33 @@ type ProcessEntity struct {
 type WorkloadMetaExtractor struct {
 	// Cache is a map from process hash to the workloadmeta entity
 	// The cache key takes the form of `pid:<pid>|createTime:<createTime>`. See hashProcess
-	cache      map[string]*ProcessEntity
-	cacheMutex sync.RWMutex
+	cache        map[string]*ProcessEntity
+	cacheVersion int32
+	cacheMutex   sync.RWMutex
 
-	grpcListener mockableGrpcListener
```

Comment:
`Extract` was calling `w.grpcListener.writeEvents(deadProcs, newEntities)` which keeps the Extractor and the gRPC server coupled.

I've opted to separate them and pass cache diffs through a channel that the gRPC server consumes from. This way the data path is a bit clearer and prevents degrading the process check execution if `writeEvents` is slow.